packages/loopring_v3/contracts/amm/LoopringAmmPool.sol

```diff
@@ -119,4 +130,22 @@ contract LoopringAmmPool is
     {
         state.withdrawWhenOffline();
     }
+
+    function canDrain(address drainer, address token)
+        public
+        override
+        view
+        returns (bool)
+    {
+        if (drainer != state.exchange.owner()) {
+            return false;
+        }
+
+        for (uint i = 0; i < state.tokens.length; i++) {
+            if (token == state.tokens[i].addr) {
+                return false;
+            }
+        }
```

Comment:
I think this needs to be like this:
```
// Never allow draining the pool token, these can be stored in this contract for forced exits at all times
if (token == address(this)) {
    return false;
}

// Allow draining the tokens of this pool only when the pool is online,
// when offline those tokens can be stored in this contract for use in withdrawWhenOffline
if (!S.isOnline()) {
    for (uint i = 0; i < state.tokens.length; i++) {
        if (token == state.tokens[i].addr) {
            return false;
        }
    }
}
```

Reply:
Based on your description, it should be like this?

```solidity
    {
        if (drainer != state.exchange.owner() || !S.isOnline()) {
            return false;
        }

        // Allow draining the tokens of this pool only when the pool is online,
        // when offline those tokens can be stored in this contract for use in withdrawWhenOffline
        for (uint i = 0; i < state.tokens.length; i++) {
            if (token == state.tokens[i].addr) {
                return false;
            }
        }

        return true;
    }
```